common/headtracker/types/head_saver.go

```diff
@@ -0,0 +1,21 @@
+package types
+
+import (
+	"context"
+
+	commontypes "github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// HeadSaver is an chain agnostic interface for saving and loading heads
+// Each chain will have its own custom implementation of HeadSaver interface.
+type HeadSaver[H commontypes.Head[BLOCK_HASH], BLOCK_HASH commontypes.Hashable] interface {
+	// Save updates the latest block number, if indeed the latest, and persists
+	// this number in case of reboot.
+	Save(ctx context.Context, head H) error
+	// Load loads latest EvmHeadTrackerHistoryDepth heads, returns the latest chain.
+	Load(ctx context.Context) (H, error)
```

Comment:
Side note: if they finality tag proposal moves forward, this might have to be changed to the latest to latest finalized instead of HistoryDepth.

Reply:
thanks for the note @dimriou !